zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaalTaakEventListener.kt

```diff
@@ -0,0 +1,178 @@
+) {
+
+    @EventListener(NotificatiesApiNotificationReceivedEvent::class)
+    fun processCompletePortaalTaakEvent(event: NotificatiesApiNotificationReceivedEvent) {
+        val objectType = event.kenmerken["objectType"]
+
+        if (!event.kanaal.equals("objecten", ignoreCase = true) ||
+            !event.actie.equals("update", ignoreCase = true) ||
+            objectType == null
+        ) {
+            return
+        }
+
+        val objectManagement =
+            objectManagementService.findByObjectTypeId(objectType.substringAfterLast("/")) ?: return
+
+        // todo need clarification why this is really needed
+        pluginService.createInstance(PortaaltaakPlugin::class.java) { properties: JsonNode
+            ->
+            properties.get("objectManagementConfigurationId").textValue().equals(objectManagement.id.toString())
+        }?.run {
+            val taakObject: TaakObject =
+                jacksonObjectMapper().convertValue(getPortaalTaakObjectData(objectManagement, event))
+            when (taakObject.status) {
+                TaakStatus.INGEDIEND -> {
+                    val task = taskService.createTaskQuery().taskId(taakObject.verwerkerTaakId).singleResult()
+                    saveDataInDocument(taakObject, task.processInstanceId, task)
+                    startProcessToUploadDocuments(
+                        taakObject,
+                        this.uploadedDocumentsHandlerProcess,
+                        task.processInstanceId,
+                        objectManagement.objectenApiPluginConfigurationId.toString(),
+                        event.resourceUrl
+                    )
+                }
+
+                else -> throw NotificatiesException("", HttpStatus.INTERNAL_SERVER_ERROR)
+            }
+        }
+    }
+
+    internal fun saveDataInDocument(taakObject: TaakObject, processInstanceId: String, task: Task) {
+        val document =
+            processDocumentService.getDocument(CamundaProcessInstanceId(processInstanceId), task as TaskEntity)
+        documentService.modifyDocument(
+            ModifyDocumentRequest(
+                document.id().toString(),
+                jacksonObjectMapper().valueToTree(taakObject.verzondenData),
+                document.version().toString()
+            )
+        ).also { result ->
+            if (result.errors().size > 0) {
+                throw NotificatiesException(
+                    "Could not update document" +
+                            "Reason:\n" +
+                            result.errors().joinToString(separator = "\n - "),
+                    HttpStatus.INTERNAL_SERVER_ERROR
+                )
+            }
+        }
+    }
+
+    internal fun getDocumentenUrls(verzondenData: JsonNode): List<String> {
+        if (verzondenData.isMissingNode || verzondenData.isNull) {
+            return emptyList()
+        }
+        if (!verzondenData.isArray) {
+            throw RuntimeException("Not an array: '/documenten'")
```

Comment:
Maybe introduce a valtimo specific TypeMismatchException here. RuntimeException is very broad given how specific this method is.